File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>

bool check_argc(int a);
bool check_key(int a);


int main(int argc, string argv[])
{
    
    
    if (check_argc(argc) ==true)            // check if argc = 2
    {
        
        int k = atoi(argv[1]);               // k is key,   To covert k from a string to an int
    
    if (check_key(k)==true)                 // check if key is valid (not less than zero)
    {
        printf("plaintext: ");
        string p = get_string();             // ask user for an input (the plaintext)
        
        printf("ciphertext: ");
         for (int i = 0, n = strlen(p); i<n; i++)
         {
             if (isalpha(p[i]))              // check if the i'th number in plaintext is alphabetic
             {
                if (islower(p[i]))            // check if the letter is small
                {
                  int char_num = ((p[i]-'a')+k)%26;     // To transform from ascii to alphabet
                    char_num +='a';                     // To transform from alphabet to ascii
                    printf("%c", char_num);
                }
                else if (isupper(p[i]))           // check if the letter is capital
                {
                    int char_num = ((p[i]-'A')+k)%26;       // like in line 30
                    char_num +='A';
                    printf("%c", char_num);
                }
             }
             else                       // to print char if it's not alphabetic
             {
                 printf("%c", p[i]);
             }
         }
        printf("\n");
    }
    }
    else
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
}


bool check_argc(int a)                   // check if argc is valid
{
    if (a == 2)
    {
        return true;
    }
    else
    {
        return false;
    }
}

bool check_key(int a)
{
    if (a >= 0)
    {
        return true;
    }
    else
    {
        return false;
    }
}

File: caesar.py
import cs50
import sys
# "str." for isalpha, isspace, ex.: str.islower

def main():
    if len(sys.argv) != 2:
        print("python caesar.py key")
        return 1
    key = int(sys.argv[1])
    if key < 0:
        print("key must be non-negative int")
        return 2
    print("Plaintext: ", end="")
    plain = cs50.get_string()
    print("Ciphertext: ", end="")
    for c in plain:
        if str.isspace(c) or str.isdigit(c):
            print("{}".format(c), end="")
        elif str.islower(c):
            encipher = ((ord(c) - ord('a')) + key) % 26
            c = chr(encipher + ord('a'))
            print(c, end="")
        else:
            encipher = ((ord(c) - ord('A')) + key) % 26
            c = chr(encipher + ord('A'))
            print("{}".format(c), end="")
    print()
    return 0

if __name__=="__main__":
    main()